woocommerce/woocommerce-ios
# Product Creation AI: Add view for generating product name

Closes: #10722 


## Description

This PR adds a new screen for generating product names from provided keywords. The screen has not been integrated yet so it can only be tested through SwiftUI preview.

## Testing instructions

Please update the `ProductNameGenerationViewModel` file's properties: `keywords`, `suggestedText`, `generationInProgress`, and `errorMessage` to see what the view looks like in different states.

## Screenshots

| Initial | With keywords | With suggested text |
| ----- | ----- | ----- |
| <img width="244" alt="Screenshot 2023-09-18 at 17 55 50" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/5bd6da30-4c63-4d84-be83-9bbbe12b55c9"> | <img width="238" alt="Screenshot 2023-09-18 at 17 56 22" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/55ed166a-4de4-4c7e-899b-0c81cbd31e6d"> | <img width="237" alt="Screenshot 2023-09-18 at 17 59 08" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/eb231919-9237-47e2-8f21-1381b9e3c5bd"> |

| Initial loading state | Loading state with suggested text | 
| ----- | ----- | 
| <img width="217" alt="Screenshot 2023-09-19 at 09 31 22" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/79b70c31-3811-4df0-8c25-fe669386b4c2"> | <img width="220" alt="Screenshot 2023-09-19 at 09 32 39" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/6d831159-be6b-4d7a-ab3c-c8acc1cefb21"> |


---
